### ISSUE TITLE
As a player browsing the library, I click the "AI" tab expecting AI libraries — but every card in that tab shows a "Deep Learning" badge instead

### ISSUE THREAD

**Issue by BenjaminSRussell** [BenjaminSRussell/Data_science_dev#1642]
As a player, I open the Library screen and see four category tabs. One of them, `data-cat="ai"`, is labeled "AI" (`index.html:1030`: `<button class="lib-cat-btn" data-cat="ai"> AI</button>`).

I click it, expecting to see AI-related libraries (TensorFlow, PyTorch) labeled consistently with the tab I just clicked. Instead, both cards render a badge that says "Deep Learning", because `src/js/game/LibraryDatabase.js:83` defines:
```js
ai: ' Deep Learning'
```
and `src/js/ui/UIUpdater.js:508` renders that value directly onto each card:
```js
<span class="lib-cat-badge">${CATEGORIES[lib.category] || lib.category}</span>
```
The filtering itself works correctly (it filters on the raw `category: 'ai'` id, not the display label), so this is purely a labeling inconsistency — but as a player it reads as if I clicked the wrong tab, or the game doesn't know its own taxonomy. "AI" and "Deep Learning" are being used interchangeably for the same category id in two different UI surfaces that sit right next to each other.

Fix direction: pick one label and use it in both places — either change `CATEGORIES.ai` to `'AI'` to match the tab button, or change the tab button text at `index.html:1030` to `"Deep Learning"` to match the badge.

---
_Difficulty score: 7/25. Prompt style: user-story. Type: bugfix._

@ollama-bot please fix this issue.
